--- origsrc/freerdp-0.8.2/channels/cliprdr/Makefile.am	2010-11-08 22:36:30.000000000 -0600
+++ src/freerdp-0.8.2/channels/cliprdr/Makefile.am	2011-06-24 01:40:36.451229700 -0500
@@ -10,7 +10,7 @@ cliprdr_la_SOURCES = \
 
 cliprdr_la_CFLAGS = -I$(top_srcdir)/include -I$(srcdir)/../common
 
-cliprdr_la_LDFLAGS = -avoid-version -module
+cliprdr_la_LDFLAGS = -avoid-version -module -no-undefined
 
 cliprdr_la_LIBADD = ../common/libcommon.la
 
--- origsrc/freerdp-0.8.2/channels/drdynvc/Makefile.am	2010-11-12 10:03:14.000000000 -0600
+++ src/freerdp-0.8.2/channels/drdynvc/Makefile.am	2011-06-24 01:41:17.546364600 -0500
@@ -13,7 +13,7 @@ drdynvc_la_SOURCES = \
 drdynvc_la_CFLAGS = -I$(top_srcdir)/include -I$(srcdir)/../common -I. \
 	-DPLUGIN_PATH=\"$(PLUGIN_PATH)\"
 
-drdynvc_la_LDFLAGS = -avoid-version -module
+drdynvc_la_LDFLAGS = -avoid-version -module -no-undefined
 
 drdynvc_la_LIBADD = ../common/libcommon.la
 
--- origsrc/freerdp-0.8.2/channels/drdynvc/audin/Makefile.am	2010-11-08 22:36:30.000000000 -0600
+++ src/freerdp-0.8.2/channels/drdynvc/audin/Makefile.am	2011-06-24 02:31:24.837634800 -0500
@@ -11,7 +11,7 @@ audin_la_SOURCES = \
 audin_la_CFLAGS = -I$(top_srcdir)/include -I$(srcdir)/../../common \
 	-I$(srcdir)/.. -DPLUGIN_PATH=\"$(PLUGIN_PATH)\"
 
-audin_la_LDFLAGS = -avoid-version -module
+audin_la_LDFLAGS = -avoid-version -module -no-undefined
 
 audin_la_LIBADD = ../../common/libcommon.la
 
--- origsrc/freerdp-0.8.2/channels/drdynvc/pnpdr/Makefile.am	2010-11-08 22:36:30.000000000 -0600
+++ src/freerdp-0.8.2/channels/drdynvc/pnpdr/Makefile.am	2011-06-24 01:41:23.066680300 -0500
@@ -11,7 +11,7 @@ pnpdr_la_SOURCES = \
 pnpdr_la_CFLAGS = -I$(top_srcdir)/include -I$(srcdir)/../../common \
 	-I$(srcdir)/.. -DPLUGIN_PATH=\"$(PLUGIN_PATH)\"
 
-pnpdr_la_LDFLAGS = -avoid-version -module
+pnpdr_la_LDFLAGS = -avoid-version -module -no-undefined
 
 pnpdr_la_LIBADD = ../../common/libcommon.la
 
--- origsrc/freerdp-0.8.2/channels/rdpdr/Makefile.am	2010-11-12 10:03:14.000000000 -0600
+++ src/freerdp-0.8.2/channels/rdpdr/Makefile.am	2011-06-24 01:41:31.547562000 -0500
@@ -17,7 +17,7 @@ rdpdr_la_SOURCES = \
 rdpdr_la_CFLAGS = -I$(top_srcdir)/include -I$(srcdir)/../common -I. \
 	-DPLUGIN_PATH=\"$(PLUGIN_PATH)\"
 
-rdpdr_la_LDFLAGS = -avoid-version -module
+rdpdr_la_LDFLAGS = -avoid-version -module -no-undefined
 
 rdpdr_la_LIBADD = ../common/libcommon.la
 
--- origsrc/freerdp-0.8.2/channels/rdpdr/disk/Makefile.am	2010-11-08 22:36:30.000000000 -0600
+++ src/freerdp-0.8.2/channels/rdpdr/disk/Makefile.am	2011-06-24 01:41:27.470328800 -0500
@@ -11,7 +11,7 @@ disk_la_SOURCES = \
 disk_la_CFLAGS = -I$(top_srcdir)/include -I$(srcdir)/../../common \
 	-I$(srcdir)/.. -DPLUGIN_PATH=\"$(PLUGIN_PATH)\"
 
-disk_la_LDFLAGS = -avoid-version -module
+disk_la_LDFLAGS = -avoid-version -module -no-undefined
 
 disk_la_LIBADD = ../../common/libcommon.la
 
--- origsrc/freerdp-0.8.2/channels/rdpdr/parallel/Makefile.am	2010-11-08 22:36:30.000000000 -0600
+++ src/freerdp-0.8.2/channels/rdpdr/parallel/Makefile.am	2011-06-24 01:41:35.462785900 -0500
@@ -11,7 +11,7 @@ parallel_la_SOURCES = \
 parallel_la_CFLAGS = -I$(top_srcdir)/include -I$(srcdir)/../../common \
 	-I$(srcdir)/.. -DPLUGIN_PATH=\"$(PLUGIN_PATH)\"
 
-parallel_la_LDFLAGS = -avoid-version -module
+parallel_la_LDFLAGS = -avoid-version -module -no-undefined
 
 parallel_la_LIBADD = ../../common/libcommon.la
 
--- origsrc/freerdp-0.8.2/channels/rdpdr/printer/Makefile.am	2010-11-08 22:36:30.000000000 -0600
+++ src/freerdp-0.8.2/channels/rdpdr/printer/Makefile.am	2011-06-24 02:38:29.138841600 -0500
@@ -11,7 +11,7 @@ printer_la_SOURCES = \
 printer_la_CFLAGS = -I$(top_srcdir)/include -I$(srcdir)/../../common \
 	-I$(srcdir)/.. -DPLUGIN_PATH=\"$(PLUGIN_PATH)\"
 
-printer_la_LDFLAGS = -avoid-version -module
+printer_la_LDFLAGS = -avoid-version -module -no-undefined
 
 printer_la_LIBADD = ../../common/libcommon.la
 
--- origsrc/freerdp-0.8.2/channels/rdpdr/serial/Makefile.am	2010-11-08 22:36:30.000000000 -0600
+++ src/freerdp-0.8.2/channels/rdpdr/serial/Makefile.am	2011-06-24 01:41:43.172623400 -0500
@@ -11,7 +11,7 @@ serial_la_SOURCES = \
 serial_la_CFLAGS = -I$(top_srcdir)/include -I$(srcdir)/../../common \
 	-I$(srcdir)/.. -DPLUGIN_PATH=\"$(PLUGIN_PATH)\"
 
-serial_la_LDFLAGS = -avoid-version -module
+serial_la_LDFLAGS = -avoid-version -module -no-undefined
 
 serial_la_LIBADD = ../../common/libcommon.la
 
--- origsrc/freerdp-0.8.2/channels/rdpsnd/Makefile.am	2010-11-12 10:03:14.000000000 -0600
+++ src/freerdp-0.8.2/channels/rdpsnd/Makefile.am	2011-06-24 01:41:47.105244900 -0500
@@ -14,7 +14,7 @@ rdpsnd_la_SOURCES = \
 rdpsnd_la_CFLAGS = -I$(top_srcdir)/include -I$(srcdir)/../common \
 	-DPLUGIN_PATH=\"$(PLUGIN_PATH)\"
 
-rdpsnd_la_LDFLAGS = -avoid-version -module
+rdpsnd_la_LDFLAGS = -avoid-version -module -no-undefined
 
 rdpsnd_la_LIBADD = ../common/libcommon.la
 
--- origsrc/freerdp-0.8.2/libfreerdp/Makefile.am	2010-11-12 10:03:14.000000000 -0600
+++ src/freerdp-0.8.2/libfreerdp/Makefile.am	2011-06-24 01:39:52.027103300 -0500
@@ -64,7 +64,7 @@ endif
 libfreerdp_la_CFLAGS = -I$(top_srcdir) -I$(top_srcdir)/include -I$(top_srcdir)/include/freerdp \
 	-I$(top_srcdir)/asn1 @CRYPTO_CFLAGS@ -DFREERDP_EXPORTS
 
-libfreerdp_la_LDFLAGS =
+libfreerdp_la_LDFLAGS = -no-undefined
 
 libfreerdp_la_LIBADD = ../asn1/libasn1.la @CRYPTO_LIBS@ @LIBICONV@
 
--- origsrc/freerdp-0.8.2/libfreerdpchanman/Makefile.am	2010-06-30 22:13:58.000000000 -0500
+++ src/freerdp-0.8.2/libfreerdpchanman/Makefile.am	2011-06-24 01:40:01.632247500 -0500
@@ -11,7 +11,7 @@ libfreerdpchanman_la_SOURCES = \
 libfreerdpchanman_la_CFLAGS = -I$(top_srcdir) -I$(top_srcdir)/include \
 	-DPLUGIN_PATH=\"$(PLUGIN_PATH)\"
 
-libfreerdpchanman_la_LDFLAGS = -ldl -lpthread
+libfreerdpchanman_la_LDFLAGS = -ldl -lpthread -no-undefined
 
 libfreerdpchanman_la_LIBDADD = ../libfreerdp/libfreerdp.la
 
--- origsrc/freerdp-0.8.2/libfreerdpgdi/Makefile.am	2010-11-08 22:36:30.000000000 -0600
+++ src/freerdp-0.8.2/libfreerdpgdi/Makefile.am	2011-06-24 01:40:07.745993700 -0500
@@ -12,7 +12,7 @@ libfreerdpgdi_la_SOURCES = \
 
 libfreerdpgdi_la_CFLAGS = -I$(top_srcdir) -I$(top_srcdir)/include
 
-libfreerdpgdi_la_LDFLAGS =
+libfreerdpgdi_la_LDFLAGS = -no-undefined
 
 # extra
 EXTRA_DIST =
--- origsrc/freerdp-0.8.2/libfreerdpkbd/Makefile.am	2010-06-27 20:21:22.000000000 -0500
+++ src/freerdp-0.8.2/libfreerdpkbd/Makefile.am	2011-06-24 01:40:12.168246600 -0500
@@ -13,7 +13,7 @@ libfreerdpkbd_la_SOURCES = \
 libfreerdpkbd_la_CFLAGS = -I$(top_srcdir) -I$(top_srcdir)/include \
 	-DKEYMAP_PATH=\"$(KEYMAP_PATH)\"
 
-libfreerdpkbd_la_LDFLAGS =
+libfreerdpkbd_la_LDFLAGS = -no-undefined
 
 # extra
 EXTRA_DIST =
