--- origsrc/freerdp-0.8.2/include/freerdp/chanman.h	2010-11-12 10:03:14.000000000 -0600
+++ src/freerdp-0.8.2/include/freerdp/chanman.h	2011-06-24 03:23:50.225747100 -0500
@@ -26,7 +26,7 @@
 
 #include <freerdp/freerdp.h>
 
-#if defined _WIN32 || defined __CYGWIN__
+#if defined _WIN32
   #ifdef FREERDP_CHANMAN_EXPORTS
     #ifdef __GNUC__
       #define FREERDP_CHANMAN_API __attribute__((dllexport))
--- origsrc/freerdp-0.8.2/include/freerdp/freerdp.h	2010-11-12 10:03:14.000000000 -0600
+++ src/freerdp-0.8.2/include/freerdp/freerdp.h	2011-06-24 03:22:38.097209700 -0500
@@ -29,7 +29,7 @@
 
 #define FREERDP_INTERFACE_VERSION 3
 
-#if defined _WIN32 || defined __CYGWIN__
+#if defined _WIN32
   #ifdef FREERDP_EXPORTS
     #ifdef __GNUC__
       #define FREERDP_API __attribute__((dllexport))
--- origsrc/freerdp-0.8.2/libfreerdpchanman/libfreerdpchanman.c	2010-11-08 22:36:30.000000000 -0600
+++ src/freerdp-0.8.2/libfreerdpchanman/libfreerdpchanman.c	2011-06-24 01:56:00.237009400 -0500
@@ -74,7 +74,11 @@
 #define DLSYM(f, n) dlsym(f, n)
 #define DLCLOSE(f) dlclose(f)
 #define PATH_SEPARATOR '/'
+#ifdef __CYGWIN__
+#define PLUGIN_EXT "dll"
+#else
 #define PLUGIN_EXT "so"
+#endif
 #define STRCHR strchr
 #endif
 #include <freerdp/freerdp.h>
