--- origsrc/freerdp-0.8.2/channels/drdynvc/audin/Makefile.am	2010-11-08 22:36:30.000000000 -0600
+++ src/freerdp-0.8.2/channels/drdynvc/audin/Makefile.am	2011-06-24 02:31:24.837634800 -0500
@@ -1,5 +1,7 @@
 ## Process this file with automake to produce Makefile.in
 
+if WITH_RDPSND
+
 # audin
 audindir = $(PLUGIN_PATH)
 
@@ -26,3 +28,4 @@ EXTRA_DIST =
 
 DISTCLEANFILES = 
 
+endif
--- origsrc/freerdp-0.8.2/channels/rdpdr/printer/Makefile.am	2010-11-08 22:36:30.000000000 -0600
+++ src/freerdp-0.8.2/channels/rdpdr/printer/Makefile.am	2011-06-24 02:38:29.138841600 -0500
@@ -1,5 +1,7 @@
 ## Process this file with automake to produce Makefile.in
 
+if RDPDR_PRINTER_CUPS
+
 # printer
 printerdir = $(PLUGIN_PATH)
 
@@ -15,7 +17,6 @@ printer_la_SOURCES = \
 
 printer_la_LIBADD = ../../common/libcommon.la
 
-if RDPDR_PRINTER_CUPS
 printer_la_SOURCES += printer_cups.c
 printer_la_CFLAGS += @CUPS_CFLAGS@
 printer_la_LIBADD += @CUPS_LIBS@
