--- origsrc/FreeRDP-1.0.2/channels/rdpdr/serial/serial_main.c	2013-01-02 15:46:59.000000000 -0600
+++ src/FreeRDP-1.0.2/channels/rdpdr/serial/serial_main.c	2014-07-10 23:52:04.497274100 -0500
@@ -22,6 +22,7 @@
 #include <stdlib.h>
 #include <string.h>
 #include <errno.h>
+#include <sys/time.h>
 
 #include "config.h"
 
--- origsrc/FreeRDP-1.0.2/include/freerdp/api.h	2013-01-02 15:46:59.000000000 -0600
+++ src/FreeRDP-1.0.2/include/freerdp/api.h	2014-07-10 23:52:04.503274900 -0500
@@ -22,7 +22,7 @@
 
 #define FREERDP_INTERFACE_VERSION 4
 
-#if defined _WIN32 || defined __CYGWIN__
+#if defined _WIN32
   #ifdef FREERDP_EXPORTS
     #ifdef __GNUC__
       #define FREERDP_API __attribute__((dllexport))
