--- origsrc/FreeRDP-1.0.2/CMakeLists.txt	2013-01-02 15:46:59.000000000 -0600
+++ src/FreeRDP-1.0.2/CMakeLists.txt	2014-07-11 00:02:54.743844900 -0500
@@ -126,10 +126,10 @@ endif()
 test_big_endian(BIG_ENDIAN)
 
 # Path to put keymaps
-set(FREERDP_KEYMAP_PATH "${CMAKE_INSTALL_PREFIX}/freerdp/keymaps")
+set(FREERDP_KEYMAP_PATH "${CMAKE_INSTALL_PREFIX}/share/freerdp-${FREERDP_VERSION}/keymaps")
 
 # Path to put plugins
-set(FREERDP_PLUGIN_PATH "${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}/freerdp")
+set(FREERDP_PLUGIN_PATH "${CMAKE_INSTALL_PREFIX}/${CMAKE_INSTALL_LIBDIR}/freerdp-${FREERDP_VERSION}")
 
 # Include directories
 include_directories(${CMAKE_CURRENT_BINARY_DIR})
--- origsrc/FreeRDP-1.0.2/keymaps/CMakeLists.txt	2013-01-02 15:46:59.000000000 -0600
+++ src/FreeRDP-1.0.2/keymaps/CMakeLists.txt	2014-07-11 00:04:03.504076400 -0500
@@ -1,3 +1,3 @@
-install(FILES aliases amiga ataritt empty evdev fujitsu hp ibm macintosh macosx sony sun xfree86 xfree98 xkb.pl DESTINATION share/freerdp/keymaps)
-install(DIRECTORY digital_vndr DESTINATION share/freerdp/keymaps FILES_MATCHING PATTERN "*")
-install(DIRECTORY sgi_vndr DESTINATION share/freerdp/keymaps FILES_MATCHING PATTERN "*")
+install(FILES aliases amiga ataritt empty evdev fujitsu hp ibm macintosh macosx sony sun xfree86 xfree98 xkb.pl DESTINATION ${FREERDP_KEYMAP_PATH})
+install(DIRECTORY digital_vndr DESTINATION ${FREERDP_KEYMAP_PATH} FILES_MATCHING PATTERN "*")
+install(DIRECTORY sgi_vndr DESTINATION ${FREERDP_KEYMAP_PATH} FILES_MATCHING PATTERN "*")
--- origsrc/FreeRDP-1.0.2/libfreerdp-cache/CMakeLists.txt	2013-01-02 15:46:59.000000000 -0600
+++ src/FreeRDP-1.0.2/libfreerdp-cache/CMakeLists.txt	2014-07-10 23:54:58.603382800 -0500
@@ -28,9 +28,12 @@ set(FREERDP_CACHE_SRCS
 
 add_library(freerdp-cache ${FREERDP_CACHE_SRCS})
 
-set_target_properties(freerdp-cache PROPERTIES VERSION ${FREERDP_VERSION_FULL} SOVERSION ${FREERDP_VERSION} PREFIX "lib")
+set_target_properties(freerdp-cache PROPERTIES VERSION ${FREERDP_VERSION_FULL} SOVERSION ${FREERDP_VERSION})
 
 target_link_libraries(freerdp-cache freerdp-core)
 target_link_libraries(freerdp-cache freerdp-utils)
 
-install(TARGETS freerdp-cache DESTINATION ${CMAKE_INSTALL_LIBDIR})
+install(TARGETS freerdp-cache
+	LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+	RUNTIME DESTINATION bin
+	ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
--- origsrc/FreeRDP-1.0.2/libfreerdp-channels/CMakeLists.txt	2013-01-02 15:46:59.000000000 -0600
+++ src/FreeRDP-1.0.2/libfreerdp-channels/CMakeLists.txt	2014-07-10 23:55:07.196474000 -0500
@@ -25,8 +25,10 @@ set(FREERDP_CHANNELS_SRCS
 
 add_library(freerdp-channels ${FREERDP_CHANNELS_SRCS})
 
-set_target_properties(freerdp-channels PROPERTIES VERSION ${FREERDP_VERSION_FULL} SOVERSION ${FREERDP_VERSION} PREFIX "lib")
+set_target_properties(freerdp-channels PROPERTIES VERSION ${FREERDP_VERSION_FULL} SOVERSION ${FREERDP_VERSION})
 target_link_libraries(freerdp-channels freerdp-utils)
 
-install(TARGETS freerdp-channels DESTINATION ${CMAKE_INSTALL_LIBDIR})
-
+install(TARGETS freerdp-channels
+	LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+	RUNTIME DESTINATION bin
+	ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
--- origsrc/FreeRDP-1.0.2/libfreerdp-codec/CMakeLists.txt	2013-01-02 15:46:59.000000000 -0600
+++ src/FreeRDP-1.0.2/libfreerdp-codec/CMakeLists.txt	2014-07-10 23:55:51.469595900 -0500
@@ -59,8 +59,11 @@ endif()
 
 add_library(freerdp-codec ${FREERDP_CODEC_SRCS})
 
-set_target_properties(freerdp-codec PROPERTIES VERSION ${FREERDP_VERSION_FULL} SOVERSION ${FREERDP_VERSION} PREFIX "lib")
+set_target_properties(freerdp-codec PROPERTIES VERSION ${FREERDP_VERSION_FULL} SOVERSION ${FREERDP_VERSION})
 target_link_libraries(freerdp-codec freerdp-utils)
 
-install(TARGETS freerdp-codec DESTINATION ${CMAKE_INSTALL_LIBDIR})
+install(TARGETS freerdp-codec
+	LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+	RUNTIME DESTINATION bin
+	ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
 
--- origsrc/FreeRDP-1.0.2/libfreerdp-core/CMakeLists.txt	2013-01-02 15:46:59.000000000 -0600
+++ src/FreeRDP-1.0.2/libfreerdp-core/CMakeLists.txt	2014-07-10 23:55:57.448355100 -0500
@@ -96,7 +96,7 @@ set(LIBFREERDP_CORE_SRCS
 
 add_library(freerdp-core ${LIBFREERDP_CORE_SRCS})
 
-set_target_properties(freerdp-core PROPERTIES VERSION ${FREERDP_VERSION_FULL} SOVERSION ${FREERDP_VERSION} PREFIX "lib")
+set_target_properties(freerdp-core PROPERTIES VERSION ${FREERDP_VERSION_FULL} SOVERSION ${FREERDP_VERSION})
 
 if(WIN32)
 	target_link_libraries(freerdp-core ws2_32)
@@ -108,4 +108,7 @@ target_link_libraries(freerdp-core ${OPE
 target_link_libraries(freerdp-core freerdp-utils)
 target_link_libraries(freerdp-core freerdp-codec)
 
-install(TARGETS freerdp-core DESTINATION ${CMAKE_INSTALL_LIBDIR})
+install(TARGETS freerdp-core
+	LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+	RUNTIME DESTINATION bin
+	ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
--- origsrc/FreeRDP-1.0.2/libfreerdp-gdi/CMakeLists.txt	2013-01-02 15:46:59.000000000 -0600
+++ src/FreeRDP-1.0.2/libfreerdp-gdi/CMakeLists.txt	2014-07-10 23:55:59.653635200 -0500
@@ -42,6 +42,9 @@ target_link_libraries(freerdp-gdi freerd
 target_link_libraries(freerdp-gdi freerdp-cache)
 target_link_libraries(freerdp-gdi freerdp-codec)
 
-set_target_properties(freerdp-gdi PROPERTIES VERSION ${FREERDP_VERSION_FULL} SOVERSION ${FREERDP_VERSION} PREFIX "lib")
+set_target_properties(freerdp-gdi PROPERTIES VERSION ${FREERDP_VERSION_FULL} SOVERSION ${FREERDP_VERSION})
 
-install(TARGETS freerdp-gdi DESTINATION ${CMAKE_INSTALL_LIBDIR})
+install(TARGETS freerdp-gdi
+	LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+	RUNTIME DESTINATION bin
+	ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
--- origsrc/FreeRDP-1.0.2/libfreerdp-kbd/CMakeLists.txt	2013-01-02 15:46:59.000000000 -0600
+++ src/FreeRDP-1.0.2/libfreerdp-kbd/CMakeLists.txt	2014-07-10 23:56:01.861915600 -0500
@@ -47,7 +47,10 @@ endif()
 
 add_definitions(-DKEYMAP_PATH="${FREERDP_KEYMAP_PATH}")
 
-set_target_properties(freerdp-kbd PROPERTIES VERSION ${FREERDP_VERSION_FULL} SOVERSION ${FREERDP_VERSION} PREFIX "lib")
+set_target_properties(freerdp-kbd PROPERTIES VERSION ${FREERDP_VERSION_FULL} SOVERSION ${FREERDP_VERSION})
 
-install(TARGETS freerdp-kbd DESTINATION ${CMAKE_INSTALL_LIBDIR})
+install(TARGETS freerdp-kbd
+	LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+	RUNTIME DESTINATION bin
+	ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
 
--- origsrc/FreeRDP-1.0.2/libfreerdp-rail/CMakeLists.txt	2013-01-02 15:46:59.000000000 -0600
+++ src/FreeRDP-1.0.2/libfreerdp-rail/CMakeLists.txt	2014-07-10 23:56:04.039192100 -0500
@@ -26,8 +26,11 @@ set(FREERDP_RAIL_SRCS
 
 add_library(freerdp-rail ${FREERDP_RAIL_SRCS})
 
-set_target_properties(freerdp-rail PROPERTIES VERSION ${FREERDP_VERSION_FULL} SOVERSION ${FREERDP_VERSION} PREFIX "lib")
+set_target_properties(freerdp-rail PROPERTIES VERSION ${FREERDP_VERSION_FULL} SOVERSION ${FREERDP_VERSION})
 
 target_link_libraries(freerdp-rail freerdp-utils)
 
-install(TARGETS freerdp-rail DESTINATION ${CMAKE_INSTALL_LIBDIR})
+install(TARGETS freerdp-rail
+	LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+	RUNTIME DESTINATION bin
+	ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
--- origsrc/FreeRDP-1.0.2/libfreerdp-utils/CMakeLists.txt	2013-01-02 15:46:59.000000000 -0600
+++ src/FreeRDP-1.0.2/libfreerdp-utils/CMakeLists.txt	2014-07-10 23:56:06.296478700 -0500
@@ -53,7 +53,7 @@ add_definitions(-DPLUGIN_PATH="${FREERDP
 
 add_library(freerdp-utils ${FREERDP_UTILS_SRCS})
 
-set_target_properties(freerdp-utils PROPERTIES VERSION ${FREERDP_VERSION_FULL} SOVERSION ${FREERDP_VERSION} PREFIX "lib")
+set_target_properties(freerdp-utils PROPERTIES VERSION ${FREERDP_VERSION_FULL} SOVERSION ${FREERDP_VERSION})
 
 target_link_libraries(freerdp-utils ${CMAKE_THREAD_LIBS_INIT} ${CMAKE_DL_LIBS})
 
@@ -64,4 +64,7 @@ if(${CMAKE_SYSTEM_NAME} MATCHES SunOS)
 	target_link_libraries(freerdp-utils rt)
 endif()
 
-install(TARGETS freerdp-utils DESTINATION ${CMAKE_INSTALL_LIBDIR})
+install(TARGETS freerdp-utils
+	LIBRARY DESTINATION ${CMAKE_INSTALL_LIBDIR}
+	RUNTIME DESTINATION bin
+	ARCHIVE DESTINATION ${CMAKE_INSTALL_LIBDIR})
