--- origsrc/FreeRDP-f77b4a57dcc373e0cad9285800acb905fcbce048/winpr/include/winpr/library.h	2017-03-03 07:41:11.000000000 -0600
+++ src/FreeRDP-f77b4a57dcc373e0cad9285800acb905fcbce048/winpr/include/winpr/library.h	2017-03-16 01:42:46.790501200 -0500
@@ -60,21 +60,25 @@ WINPR_API HMODULE LoadLibraryExW(LPCWSTR
 
 #endif
 
-#if !defined(_WIN32) && !defined(__CYGWIN__)
+#if defined(_WIN32) || defined(__CYGWIN__)
+#define WINPR_LIBRARY_API __stdcall __attribute__((dllimport))
+#else
+#define WINPR_LIBRARY_API WINPR_API
+#endif
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
-WINPR_API HMODULE GetModuleHandleA(LPCSTR lpModuleName);
-WINPR_API HMODULE GetModuleHandleW(LPCWSTR lpModuleName);
+WINPR_LIBRARY_API HMODULE GetModuleHandleA(LPCSTR lpModuleName);
+WINPR_LIBRARY_API HMODULE GetModuleHandleW(LPCWSTR lpModuleName);
 
-WINPR_API DWORD GetModuleFileNameA(HMODULE hModule, LPSTR lpFilename, DWORD nSize);
-WINPR_API DWORD GetModuleFileNameW(HMODULE hModule, LPWSTR lpFilename, DWORD nSize);
+WINPR_LIBRARY_API DWORD GetModuleFileNameA(HMODULE hModule, LPSTR lpFilename, DWORD nSize);
+WINPR_LIBRARY_API DWORD GetModuleFileNameW(HMODULE hModule, LPWSTR lpFilename, DWORD nSize);
 
-WINPR_API FARPROC GetProcAddress(HMODULE hModule, LPCSTR lpProcName);
+WINPR_LIBRARY_API FARPROC GetProcAddress(HMODULE hModule, LPCSTR lpProcName);
 
-WINPR_API BOOL FreeLibrary(HMODULE hLibModule);
+WINPR_LIBRARY_API BOOL FreeLibrary(HMODULE hLibModule);
 
 #ifdef __cplusplus
 }
@@ -88,7 +92,5 @@ WINPR_API BOOL FreeLibrary(HMODULE hLibM
 #define GetModuleFileName	GetModuleFileNameA
 #endif
 
-#endif
-
 #endif /* WINPR_LIBRARY_H */
 
--- origsrc/FreeRDP-d72ff5d241c63822af9f0d6784a060129d9d010d/winpr/libwinpr/synch/synch.h	2016-12-02 15:54:06.000000000 -0600
+++ src/FreeRDP-d72ff5d241c63822af9f0d6784a060129d9d010d/winpr/libwinpr/synch/synch.h	2017-02-09 21:48:11.411736500 -0600
@@ -28,7 +28,7 @@
 
 #include <winpr/synch.h>
 
-#ifdef __linux__
+#if defined(__linux__) || defined(__CYGWIN__)
 #define WITH_POSIX_TIMER	1
 #endif
 
--- origsrc/FreeRDP-d72ff5d241c63822af9f0d6784a060129d9d010d/winpr/libwinpr/sysinfo/sysinfo.c	2016-12-02 15:54:06.000000000 -0600
+++ src/FreeRDP-d72ff5d241c63822af9f0d6784a060129d9d010d/winpr/libwinpr/sysinfo/sysinfo.c	2017-02-09 21:49:41.685112000 -0600
@@ -29,7 +29,7 @@
 #include "cpufeatures/cpu-features.h"
 #endif
 
-#if defined(__linux__)
+#if defined(__linux__) || defined(__CYGWIN__)
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <fcntl.h>
@@ -97,7 +97,7 @@ static DWORD GetNumberOfProcessors()
 #if defined(ANDROID)
 	return android_getCpuCount();
 	/* TODO: iOS */
-#elif defined(__linux__) || defined(__sun) || defined(_AIX)
+#elif defined(__linux__) || defined(__sun) || defined(_AIX) || defined(__CYGWIN__)
 	numCPUs = (DWORD) sysconf(_SC_NPROCESSORS_ONLN);
 #elif defined(__MACOSX__) || \
 	defined(__FreeBSD__) || defined(__NetBSD__) || \
